internal/handlers/pg/pgdb/pool.go

```diff
@@ -345,8 +352,19 @@ func (pgPool *Pool) DropDatabase(ctx context.Context, db string) error {
 
 // CreateCollection creates a new FerretDB collection in existing schema.
 //
-// It returns ErrAlreadyExist if table already exist, ErrTableNotExist is schema does not exist.
+// It returns:
+//  - ErrInvalidTableName - if the table name doesn't conform to restrictions.
+//  - ErrAlreadyExist - if table already exist.
+//  - ErrTableNotExist - is schema does not exist.
 func (pgPool *Pool) CreateCollection(ctx context.Context, db, collection string) error {
+	if !validateCollectionNameRe.MatchString(collection) {
+		return ErrInvalidTableName
+	}
+
+	if strings.HasPrefix(collection, collectionPrefix) {
```

Comment:
With that, I wonder if the `collectionPrefix` constant should be called something like `reservedCollectionPrefix`. But I don't insist :)